src/ResourceManager/Reservations/Commands.Reservations/Common/AzureReservationsCmdletBase.cs

```diff
@@ -21,6 +22,15 @@ public abstract class AzureReservationsCmdletBase : AzureRMCmdlet
 
         private Dictionary<string, List<string>> _defaultRequestHeaders;
 
+        protected void ValidateGuidInput(string inputValue, string inputType)
```

Comment:
@corquiri is there any reason you can't just make the parameters of type `Guid` rather than of type `string`? If the server was never able to handle a non-Guid string before, then this won't be a breaking change. This will allow you to remove all of the validation you are trying to do over the strings.

Reply:
Do you mean if I change parameter to something like
         [Parameter(Mandatory = false)]
         [ValidateNotNull]
         public Guid SubscriptionId { get; set; }

Then the powershell console will just throw if the input is not a GUID? Where is the code that parses the input from console and populate the parameter of cmdlet classes? If I can verify that the parameter validation is already being done, I can follow your suggestion and change the parameters to Guid type.

Reply:
@corquiri the PowerShell engine (not controlled by us) automatically attempts to cast the provided object to whatever type the parameter is expecting, and if it's unable to do so, it will throw an error. In this case, if the user provides a string that is not in the form of a GUID, PowerShell will throw with an error saying that it is unable to convert the given string into a GUID.